### DISCUSSION TITLE
Re-using cache data of PersistentDataset on new runs (when using Azure blob storage).

### DISCUSSION THREAD

**Discussion by jhairgallardo** [Project-MONAI/MONAI#7063] [Q&A]
Hello,
I am using `PersistentDataset` on a large dataset. Is it possible to re-use the cache data on a new run instead of creating the cache data again? I notice that, if I use the same folder where I cached the data before, it won't re-use the files; instead, it will create new files (increasing the disk memory usage of that folder). I am new at using Monai, so I am not sure if I am missing some input argument or configuration that can allow `PersistentDataset` to re-use cache data generated on previous runs. (The transformations don't change between runs). Thank you!

**Comment by ebrahimebrahim** [comment]
It is indeed supposed to re-use the data that it caches to disk. Like if the data is there in the directory that you pointed it to, then it should use that data rather than creating new files. So it sounds like something is wrong.

**Reply by jhairgallardo** [reply]
I see. My cache folder is a blob storage in azure. I will test this using a local folder instead. Maybe using the blob storage folder is causing some issues.

**Reply by wyli** [reply]
the persistent dataset finds cache content by hash of input dictionaries (in this format https://github.com/Project-MONAI/MONAI/blob/317ef1fe0c568fd92cad1e66416f9d5ad0f3a8cd/monai/data/dataset.py#L179-L182) perhaps in your use cases there are changes in the dictionaries?

**Reply by jhairgallardo** [reply]
Thank you for your prompt response! When I use a local folder in cache_dir, monai re-uses the same files from a previous run. so, it works in that case! However, it seems that it doesn't reuse the files when I am pointing to a blob storage folder and do a second run there. (The first one created the files in the blob storage folder)

@wyli yes, I am using that format. Since it works just fine when using a local folder, I think the issue is related to using a remote folder that is mounted on. It could also be related to run the code in an azure cluster. Everything works fine locally in my own PC, but when I run it on azure (and use a blob storage folder for cache_dir) it doesn't reuse the files.

**Reply by jhairgallardo** [reply]
I found the issue when using a blob storage for PersistentDataset. It doesn't re-use the files because when mounting the blob, it creates a new mounted path everytime in the azure cluster node, causing the hash to not match the new folder. It is not an issue with MONAI, but with the way mounting a folder in azure cluster instances works. I will post here if I find a work around, but I think we can close this discussion for now since it is not a MONAI's issue. Thank you!
